  This patch fixes bug #1012 where the job is canceled because
  of Max Run Time exceeded when the job has not yet started.

  Apply it to 2.2.6 or any prior 2.2.x version with:

  cd <bacula-source>
  patch -p0 <2.2.6-maxruntime.patch
  ./configure <your-options>
  make
  ...
  make install


Index: src/dird/job.c
===================================================================
--- src/dird/job.c	(revision 5979)
+++ src/dird/job.c	(working copy)
@@ -556,7 +556,7 @@
  */
 static bool job_check_maxruntime(JCR *control_jcr, JCR *jcr)
 {
-   if (jcr->job->MaxRunTime == 0 || job_canceled(jcr)) {
+   if (jcr->job->MaxRunTime == 0 || job_canceled(jcr) || jcr->JobStatus == JS_Created) {
       return false;
    }
    if ((watchdog_time - jcr->start_time) < jcr->job->MaxRunTime) {
@@ -931,6 +931,7 @@
 {
    jcr->job = job;
    jcr->JobType = job->JobType;
+   jcr->JobStatus = JS_Created;
    switch (jcr->JobType) {
    case JT_ADMIN:
    case JT_RESTORE:
